### PR TITLE
Tree: reduce test code dependency on changeset format details

### PR DESCRIPTION
**Pull request by yann-achard-MS**
## Description

This PR reduces the amount of test code that is directly dependent on the changeset format.
This PR does *not* reduce the amount of existing test coverage.

This PR also introduces `TestTree` and `Sequencer` classes to help write editing and rebasing tests with finer control over edit sequencing than integration tests allow for, but without targeting lower-level components as unit tests typically do.

### CODE REVIEW

File: packages/dds/tree/src/test/shared-tree/testTree.ts [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+export class Sequencer {
+    private seqNumber: number = 0;
+    private readonly clients = new Map<SessionId, ClientData>();
+
+    public constructor() {}
+
+    /**
+     * Sequences the given edit.
+     */
+    public order(edit: TestTreeEdit): CommittedTestTreeEdit;
```
Comment (alex-pardes) [inline packages/dds/tree/src/test/shared-tree/testTree.ts:206]: I'm not sure this flavor should also be called `order`.

**Review by alex-pardes — APPROVED** [review]
I'm really excited about these improvements!

File: packages/dds/tree/src/feature-libraries/sequence-field/sequenceFieldEditor.ts [hunk 2]
```diff
@@ -4,30 +4,53 @@
...
 export interface SequenceFieldEditor extends FieldEditor<Changeset> {
-    insert(index: number, cursor: ITreeCursor | ITreeCursor[]): Changeset;
-    delete(index: number, count: number): Changeset;
+    insert(index: number, cursor: ITreeCursor | ITreeCursor[]): Changeset<never>;
+    delete(index: number, count: number): Changeset<never>;
+    revive(
+        index: number,
+        count: number,
+        detachIndex: number,
+        revision?: RevisionTag,
```
[the remark this answers is not in the document]
Comment (yann-achard-MS) [reply packages/dds/tree/src/feature-libraries/sequence-field/sequenceFieldEditor.ts]: That was left over from copying the struct. I don't think it makes sense to have that degree of freedom in our tests.

File: packages/dds/tree/src/test/feature-libraries/sequence-field/utils.ts [hunk 3]
```diff
@@ -24,60 +24,54 @@ export const cases: {
...
+            makeAnonChange(createModifyChangeset(1, TestChange.mint([], 2))),
+        ],
+        TestChange.compose,
+    ),
+    delete: createDeleteChangeset(1, 3),
+    revive: createReviveChangeset(2, 2, 0, tag),
 };
 
-export function createInsertChangeset(index: number, size: number): TestChangeset {
+export function createInsertChangeset(
```
[the remark this answers is not in the document]
Comment (yann-achard-MS) [reply packages/dds/tree/src/test/feature-libraries/sequence-field/utils.ts]: Just using naming them `insert`, `delete`, etc. caused two issues:
1. A lot of tests use those names for local variables.
2. `delete` is a language keyword that can only be used as a field key.

I've made some changes to try to make things a little nicer, but feel free to suggest some other approach.

**Review by alex-pardes — APPROVED** [review]

**Pull request merged**